datadog_checks_base/datadog_checks/base/ddyaml.py

```diff
@@ -20,6 +23,23 @@
 pyyaml_dump_all = None
 
 
+def run_once(f):
+    """Runs a function (successfully) only once.
+    The running can be reset by setting the `has_run` attribute to False
+    """
+    @wraps(f)
+    def wrapper(*args, **kwargs):
+        if not wrapper.has_run:
+            wrapper.has_run = True
+            return f(*args, **kwargs)
+    wrapper.has_run = False
+    return wrapper
```

Comment:
Seems python loggers have a filter feature, is that a possible solution for this case ? seems a bit cleaner than wrapping/decorating ourself.

https://stackoverflow.com/a/31953563

```python
class DuplicateFilter(object):
    def __init__(self):
        self.msgs = set()

    def filter(self, record):
        rv = record.msg not in self.msgs
        self.msgs.add(record.msg)
        return rv

dup_filter = DuplicateFilter()
logger.addFilter(dup_filter)
for i in range(10):
    canned_example()
logger.removeFilter(dup_filter)
```

Reply:
We can forget about that, logging once was a WIP but according to different feedbacks, the best looks to be to log to debug with more information. If needed we can investigate